public/dorequest.php

```diff
@@ -264,7 +316,15 @@ function DoRequestError(string $error, ?int $status = 200, ?string $code = null)
          */
         $response = array_merge($response, unlockAchievement($user, $achIDToAward, $hardcore));
 
-        if (Achievement::where('ID', $achIDToAward)->exists()) {
+        $foundAchievement = Achievement::where('ID', $achIDToAward)->first();
+        if ($foundAchievement !== null) {
+            if (
+                $delegateTo
+                && request()->input('v') !== $foundAchievement->unlockValidationHash($foundDelegateToUser, (int) $hardcore)
```

Comment:
You may want this to be a case insensitive comparison as the validation hash can contains alphabetic characters and the client hashing algorithm may not generate the same casing the server expects. 

Alternate, we could put the onus on the client to always generate lowercase hashes. That would have to be documented somewhere.

Reply:
Agreed - the verification is now case insensitive.